processor/signozspanmetricsprocessor/processor.go

```diff
@@ -273,81 +298,90 @@ func (p *processorImp) Capabilities() consumer.Capabilities {
 // It aggregates the trace data to generate metrics, forwarding these metrics to the discovered metrics exporter.
 // The original input trace data will be forwarded to the next consumer, unmodified.
 func (p *processorImp) ConsumeTraces(ctx context.Context, traces ptrace.Traces) error {
-	// p.logger.Info("Consuming trace data")
+	// Forward trace data unmodified and propagate both metrics and trace pipeline errors, if any.
+	return multierr.Combine(p.tracesToMetrics(ctx, traces), p.nextConsumer.ConsumeTraces(ctx, traces))
+}
+
+func (p *processorImp) tracesToMetrics(ctx context.Context, traces ptrace.Traces) error {
```

Comment:
@ankitnayan this diff fixes the wonder issues since next consumer gets data even when there is an error with traceToMetrics function

Reply:
If this is a custom fix then the upstream spanmetricsprocessor would also stop forwarding traces to the next consumer on getting errors

Reply:
It was also an issue in the upstream processor, and it's the same fix we are applying.

Reply:
ok